frontend/device/kobo/powerd.lua

```diff
@@ -300,10 +300,54 @@ function KoboPowerD:isChargedHW()
     return false
 end
 
+function KoboPowerD:_postponedSetIntensityHW(end_intensity)
+    self:setIntensityHW(end_intensity)
+    self.running = false
+end
+
+function KoboPowerD:_stopFrontlightRamp()
+    -- Make sure we have no other ramp running.
+    self.UIManager:unschedule(self.turnOffFrontlightRamp)
+    self.UIManager:unschedule(self.turnOnFrontlightRamp)
+    self.UIManager:unschedule(self.setIntensityHW)
+    self.fl_rump_running = false
+end
+
+-- This ramp down goes faster on high intensity and slower at the end.
+-- The user will notice a linear brightness change.
+-- The whole function gets calles at max log(100)/log(0.75) = 17 times,
+-- which will lead to 0.025*17 + 0.5 = 0.925s ramp down time (non blocking), can be aborted.
+function BasePowerD:turnOffFrontlightRamp(curr_ramp_intensity, end_intensity)
+    curr_ramp_intensity = math.floor(math.max(curr_ramp_intensity * .75, 0))
+
+    if curr_ramp_intensity > end_intensity then
+        self:setIntensityHW(math.floor(curr_ramp_intensity))
+        self.UIManager:scheduleIn(0.025, self.turnOffFrontlightRamp, self, curr_ramp_intensity, end_intensity)
+    else
+        -- On some devices (Sage) setting intensity to zero happens immediately,
+        -- which will lead to a jump and not a ramp. So we postpone the last change by 0.5s to get it smooth
+        self.UIManager:scheduleIn(0.5, self._postponedSetIntensityHW, self, end_intensity)
+        -- no reschedule here, as we are done
```

Comment:
This will probably still need a Device check, because a lot of the older devices have the opposite issue: no matter the timing or the previous brightness level, switching from on to off incurs an extra delay, so this would make it much worse ;).

Reply:
Much, much worse, yes.